fetch/fetch.go

```diff
@@ -11,24 +11,110 @@ import (
 )
 
 var (
+	trueBool = true
+)
+
+type Fetcher struct {
 	client *ssm.SSM
 	paths  []string
 	tags   []string
+	debug  bool
+}
 
-	trueBool = true
-)
+// MustSetEnv retrieves SSM parameters for the given paths and tags and sets each
+// as env variables via os.Setenv.  This function will panic if any error occurs.
+// Set debug to true for debug logging to STDOUT (WARNING: any sensitive SSM param values will be logged).
+func MustSetEnv(paths, tags []string, debug bool) {
+	fetcher := NewFetcher(paths, tags, os.Getenv("SSM_REGION"), debug)
+	params, err := fetcher.FetchParams(paths, tags)
+	if err != nil {
+		panic(err)
+	}
 
-func init() {
+	nameValues := getParamNameValues(params)
+	for name, value := range nameValues {
+		if err := os.Setenv(name, value); err != nil {
+			panic(err)
+		}
+	}
+}
+
+// NewFetcher returns a new *Fetcher for the given paths and tags.
+// Optional ssmRegion can be supplied if the SSM region is different
+// than the AWS_REGION this program is running from.
+// Set debug to true for debug logging to STDOUT (WARNING: any sensitive SSM param values will be logged).
+func NewFetcher(paths, tags []string, ssmRegion string, debug bool) *Fetcher {
+	// initialize aws client
 	awsConfig := aws.NewConfig()
-	ssmRegion := os.Getenv("SSM_REGION")
 	if ssmRegion != "" {
 		awsConfig = awsConfig.WithRegion(ssmRegion)
 	}
 	session := session.Must(session.NewSession())
-	client = ssm.New(session, awsConfig)
+	client := ssm.New(session, awsConfig)
+
+	return &Fetcher{
+		client: client,
+		paths:  paths,
+		tags:   tags,
+		debug:  debug,
+	}
 }
 
-func FetchParams(paths, tags []string) ([]*ssm.Parameter, error) {
+func (f *Fetcher) FetchParams(paths, tags []string) ([]*ssm.Parameter, error) {
+
+	fetchByPaths := len(paths) > 0
+	fetchByTags := len(tags) > 0
+
+	// if only fetching params by tags
+	if fetchByTags && !fetchByPaths {
+		// describe all parameters with given tags
+		paramNames, err := f.describeParameters(tags)
+		if err != nil {
+			return nil, err
+		}
+
+		// fetch these parameters directly
+		params, err := f.getParameters(paramNames)
```

Comment:
By tags is essentially how GT uses this program and here is the big performance improvement. Instead of calling get-parameters-by-path using `/`, just directly get the parameters retrieved from above describe-parameters by calling get-parameters.  We'll need to update our entrypoint script and remove the `paths` param: https://github.com/gametimesf/ops/blob/master/docker/go/entrypoint.sh#L4-L6